server/src/sql-tools/schema-from-database.ts

```diff
@@ -0,0 +1,394 @@
+import { Kysely, sql } from 'kysely';
+import { PostgresJSDialect } from 'kysely-postgres-js';
+import { jsonArrayFrom } from 'kysely/helpers/postgres';
+import { Sql } from 'postgres';
+import {
+  DatabaseActionType,
+  DatabaseClient,
+  DatabaseColumn,
+  DatabaseColumnType,
+  DatabaseConstraintType,
+  DatabaseSchema,
+  DatabaseTable,
+  LoadSchemaOptions,
+  PostgresDB,
+} from 'src/sql-tools/types';
+
+/**
+ * Load the database schema from the database
+ */
+export const schemaFromDatabase = async (postgres: Sql, options: LoadSchemaOptions = {}): Promise<DatabaseSchema> => {
+  const db = createDatabaseClient(postgres);
+
+  const warnings: string[] = [];
+  const warn = (message: string) => {
+    warnings.push(message);
+  };
+
+  const schemaName = options.schemaName || 'public';
+  const tablesMap: Record<string, DatabaseTable> = {};
+
+  const [tables, columns, indexes, constraints, enums] = await Promise.all([
+    getTables(db, schemaName),
+    getTableColumns(db, schemaName),
+    getTableIndexes(db, schemaName),
+    getTableConstraints(db, schemaName),
+    getUserDefinedEnums(db, schemaName),
+  ]);
+
+  const enumMap = Object.fromEntries(enums.map((e) => [e.name, e.values]));
```

Comment:
Why not... use a map when you already call it that?

Reply:
An object of key, value pairs is by definition a map. I don't really see any value in using `new Map()` with the methods `.get()` `.set()` over plain object notation.